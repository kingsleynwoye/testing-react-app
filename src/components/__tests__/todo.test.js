import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Todo from "../todo";

afterEach(() => {
    cleanup();
});

test('Should render non-component todo', () => {
    const todo = { id: 1, title: "Leaning react testing in react", completed: false };
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("Leaning react testing in react");
    expect(todoElement).not.toContainHTML('<strike>');
});

test('Should render component todo', () => {
    const todo = { id: 2, title: "I have learn react", completed: true };
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-2");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("I have learn react");
    expect(todoElement).not.toContainHTML('<strike>');
});

test('matches snapshot', () => {
    const todo = { id: 1, title: "Leaning react testing in react", completed: false };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();
});