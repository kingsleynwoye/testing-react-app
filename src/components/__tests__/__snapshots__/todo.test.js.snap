// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    Leaning react testing in react
  </h1>
</div>
`;
